src/charm.py

```diff
@@ -134,11 +162,35 @@ def main(self, event):
                             "APP_SECURE_COOKIES": str(config["secure-cookies"]).lower(),
                             "BACKEND_MODE": config["backend-mode"],
                             "APP_PREFIX": "/volumes",
+                            "VOLUME_VIEWER_IMAGE": config["volume-viewer-image"],
                         },
                         "ports": [{"name": "http", "containerPort": config["port"]}],
+                        "volumeConfig": [
+                            {
+                                "name": "viewer-spec",
+                                "mountPath": "/etc/config/",  # xwris to .yaml?
+                                "files": [
+                                    {
+                                        "path": "viewer-spec.yaml",
+                                        "content": render_template(
+                                            "src/templates/viewer-spec.yaml.j2", {}
+                                        ),
+                                    }
+                                ],
+                            },
+                        ],
                     }
                 ],
             },
+            k8s_resources={
+                "configMaps": {
+                    "volumes-web-app-viewer-spec-ck6bhh4bdm": {
```

Comment:
Is this name going to change often?

Reply:
I wrote the reason inside the [commit message](https://github.com/canonical/kubeflow-volumes-operator/pull/115/commits/733426d6d5a183441fefd5320f0e93fe7bd03f46). Not sure if you missed this or ask for a better explanation.

>Modify pytest command to always run tests on model kubeflow. Otherwise,
they fail deterministaclly since kubeflow-dashboard requires `kubeflow`
namespace.

Reply:
That is defined in upstream manifests. I guess it will be subject to change during manifests upgrade. Didn't look into why and how this name was generated tbh.

Reply:
That is kind of weird, it looks like an auto-generated name, which usually are randomised. If it's in the upstream spec, then there is nothing much to do here, I guess.